### PR TITLE
Add a basic scrolling test for the Gallery home page.

### PR DESCRIPTION
**Pull request by eseidelGoogle**
Current numbers from my Nexus 5:
  "average_frame_build_time_millis": 22.6854,
  "missed_frame_build_budget_count": 15,
  "frame_count": 15

Filed https://github.com/flutter/flutter/issues/3296

@yjbanov @hansmuller @vlidholt

### CODE REVIEW

**Comment by sethladd** [comment]
This is awesome!!

File: examples/material_gallery/test_driver/scroll_perf_test.dart [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+        // Scroll up
+        for (int i = 0; i < 5; i++) {
+          await driver.scroll(stockList, 0.0, 300.0, new Duration(milliseconds: 300));
+          await new Future<Null>.delayed(new Duration(milliseconds: 500));
+        }
+      });
+
+      TimelineSummary summary = new TimelineSummary.summarize(timeline);
+      summary.writeSummaryToFile('home_scroll_perf', pretty: true);
```
Comment (sethladd) [inline examples/material_gallery/test_driver/scroll_perf_test.dart]: If you are looking for ways to use `..`:

```
new TimelineSummary.summarize(timeline)
  ..writeSummaryToFile(..)
  ..writeTimelineToFile(..);
```

File: examples/material_gallery/test_driver/scroll_perf_test.dart [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        ObjectRef stockList = await driver.findByValueKey('Gallery List');
+        expect(stockList, isNotNull);
+
+        await driver.tap(await driver.findByText('Demos'));
+        await driver.tap(await driver.findByText('Components'));
+        await driver.tap(await driver.findByText('Style'));
+
+        // Scroll down
+        for (int i = 0; i < 5; i++) {
+          await driver.scroll(stockList, 0.0, -300.0, new Duration(milliseconds: 300));
```
Comment (HansMuller) [inline examples/material_gallery/test_driver/scroll_perf_test.dart]: I'm not sure what the significance of this Duration is vis performance or if it corresponds to a real user's fling gesture or what the relationship between the fling velocity and the fling rate is. I think tests like this should be based on a range of experimentally determined, genuine human user, fling gesture profiles.

File: examples/material_gallery/test_driver/scroll_perf_test.dart [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          await new Future<Null>.delayed(new Duration(milliseconds: 500));
```
Comment (HansMuller) [inline examples/material_gallery/test_driver/scroll_perf_test.dart]: Not sure what we're waiting for here, not sure what the wait's significance is vis performance, not clear what 500ms is based on.

**Comment by HansMuller** [comment]
LGTM

**Comment by eseidelGoogle** [comment]
@sethladd It just felt odd to have to write all those awaits in my tap code.  I was also expecting that the tap() call might be on the ObjectRef instead of on Driver, but it's OK.  I mostly copy/pasted this one from https://github.com/flutter/flutter/blob/master/dev/benchmarks/complex_layout/test_driver/scroll_perf_test.dart

**Comment by sethladd** [comment]
@eseidel that's cool. For summary at the end, consider using `..` if you want to go "full Dart" :)

File: examples/material_gallery/test_driver/scroll_perf_test.dart (hunk 2 above)
Comment (eseidelGoogle) [inline examples/material_gallery/test_driver/scroll_perf_test.dart]: Filed https://github.com/flutter/flutter/issues/3316

File: examples/material_gallery/test_driver/scroll_perf_test.dart (hunk 3 above)
Comment (eseidelGoogle) [inline examples/material_gallery/test_driver/scroll_perf_test.dart]: Just copy/pasta from https://github.com/flutter/flutter/blob/master/dev/benchmarks/complex_layout/test_driver/scroll_perf_test.dart, happy to change.  FYI @vlidholt

Comment (yjbanov) [inline examples/material_gallery/test_driver/scroll_perf_test.dart]: And that's copypasta from my code :) If you think of scrolling as a sequence of "pointer down", "drag", "pointer up", "move finger back to the original location", "pointer down", "drag", etc, then this pause emulates the "move finger back to the original location" part. The duration of 500ms is, just like the scroll duration above it, is not based on any empirical data. It was just me testing the testing framework. Also, `scroll` does not emulate the acceleration/deceleration of a user's finger. We need to do some homework in this area.

**Pull request merged**